twisted/scripts/_twistd_unix.py

```diff
@@ -198,6 +199,38 @@ def preApplication(self):
         self.oldstderr = sys.stderr
 
 
+    def _formatException(self, exception):
+        """
+        Format the C{exception} in preparation for writing to the
+        status pipe.  This does the right thing on Python 2 if the
+        exception's message is Unicode, and in all cases limits the
+        length of the message, prior to encoding, to 100 codepoints.
+
+        This means the return message may be longer than 100 bytes.
+
+        @type exception: L{Exception}
+        @param exception: The exception to format.
+
+        @return: The formatted message, suitable for writing to the
+            status pipe.
+        @rtype: L{bytes}
+        """
+        # On Python 2 this will encode Unicode messages with the ascii
+        # codec and the backslashreplace error handler.
+        exceptionLine = traceback.format_exception_only(exception.__class__,
+                                                        exception)[-1]
+        # remove exception name
+        exceptionMessage = exceptionLine.split(': ', 1)[-1]
+        # remove the trailing newline and limit the total length to
+        # the passed string to 100
+        truncatedMessage = exceptionMessage.strip()[:98]
+        formattedMessage = '1 %s' % truncatedMessage
+        # On Python 3, ensure the message
+        if _PY3:
```

Comment:
I can't avoid it :(   The return value of `_formatException` gets written to a pipe, so it has to be bytes.  The truncated comment should read: "On Python 3, ensure the message is encoded as bytes in the same way as it is on Python 2".  So while you're right that part of the goal here is to produce consistent output, the bigger part is encoding a native Python 3 string to bytes, which is unavoidable.